packages/js/data/src/products/actions.ts

```diff
@@ -151,6 +151,48 @@ export function* updateProduct(
 	}
 }
 
+export function deleteProductSuccess(
+	id: number,
+	product: PartialProduct,
+	force: boolean
+) {
+	return {
+		type: TYPES.DELETE_PRODUCT_SUCCESS as const,
+		id,
+		product,
+		force,
+	};
+}
+
+export function deleteProductError( id: number, error: unknown ) {
+	return {
+		type: TYPES.DELETE_PRODUCT_ERROR as const,
+		id,
+		error,
+	};
+}
+
+export function* removeProduct( id: number, force = false ) {
+	try {
+		const url = force
+			? `${ WC_PRODUCT_NAMESPACE }/${ id }?force=true`
+			: `${ WC_PRODUCT_NAMESPACE }/${ id }`;
+
+		const product: Product = yield apiFetch( {
+			path: url,
+			method: 'DELETE',
+		} );
+
+		if ( product ) {
```

Comment:
Any point in wrapping this in an "if" still?  I would assume we always want to return whatever is returned from the response if it's a `200`.

Reply:
Good idea @joshuatf! I changed it in the commit 5121432
I think that this PR should be reapproved. Could you take another look?